src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobUtils.scala

```diff
@@ -59,17 +133,11 @@ object JobUtils {
   //TODO(FL): Think about moving this back into the JobScheduler, though it might be a bit crowded.
   def loadJobs(scheduler: JobScheduler, store: PersistenceStore) {
     //TODO(FL): Create functions that map strings to jobs
-    val scheduledJobs = new ListBuffer[ScheduleBasedJob]
-    val dependencyBasedJobs = new ListBuffer[DependencyBasedJob]
 
     val jobs = store.getJobs
+    val dependencyBasedJobs = jobs.collect({case d: DependencyBasedJob => d }).toList
+    val scheduledJobs= jobs.collect({ case s: InternalScheduleBasedJob => s }).toList
```

Comment:
Should I switch back to the original (mutation?). Not aware of a way to do type-safe subtype collection partitioning. Collect causes us to lose type information.